airlock_processor/shared_code/blob_operations.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+import datetime
+from azure.storage.blob import ContainerSasPermissions, generate_container_sas, BlobServiceClient
+
+from exceptions.AirlockInvalidContainerException import AirlockInvalidContainerException
+
+
+def copy_data(source_account_name: str, source_account_key: str, sa_source_connection_string: str, sa_dest_connection_string: str, request_id: str):
+    container_name = request_id
+
+    # token geneation with expiry of 1 hour. since its not shared, we can leave it to expire (no need to track/delete)
+    sas_token = generate_container_sas(account_name=source_account_name,
+                                       container_name=container_name,
+                                       account_key=source_account_key,
+                                       permission=ContainerSasPermissions(read=True),
+                                       expiry=datetime.datetime.utcnow() + datetime.timedelta(hours=1))
+
+    # Copy files
+    source_blob_service_client = BlobServiceClient.from_connection_string(sa_source_connection_string)
+    dest_blob_service_client = BlobServiceClient.from_connection_string(sa_dest_connection_string)
+
+    source_container_client = source_blob_service_client.get_container_client(container_name)
+
+    try:
+        blob_name = ""
+        for blob in source_container_client.list_blobs():
+            if blob_name != "":
+                msg = "Request with id {} contains more than 1 file. flow aborted.".format(request_id)
+                logging.error(msg)
+                raise AirlockInvalidContainerException(msg)
+            blob_name = blob.name
+
+        if blob_name == "":
```

Comment:
Same here - can we use a `if count == 0` type logic?

Reply:
can't because it is a lazy iterator

Reply:
fixed